### ISSUE TITLE
Equal validator not working

### ISSUE THREAD

**Issue by pcmvillarin** [JeremySkinner/FluentValidation#944]
Hi Sir,

I have this validation in my website application but it's not working on my end.
**Equal validator**
RuleFor(m => m.ConfirmPassword).Equal(m => m.Password).WithMessage("Password do not match");

The Other RuleFor validation works except for **Equal()**

I am using Asp.net Core Razor pages and razor tag  **asp-validation-for** to display the error message.

Thank you

**Comment by JeremySkinner** [comment]
Can you provide more details of what exactly isn't working? 

How are you executing the validator? Are the correct values appearing in ModelState?

**Comment by ozerty** [comment]
Hi Jeremy,
I think there is an issue with validations on client side that I have discovered using the "Equal()" method.
I have this actual class (cf. my ViewModel) passing to my View : 

```
public class MainViewModel
{
        public A ClassA { get; set; }

        public B ClassB { get; set; }
}
```

```
public class A
{
        public string Text1 { get; set; }

        public string Text2 { get; set; }
}
```

I apply these validations on my class A : 

```
public class ClassAValidator : AbstractValidator<A>
{
        public ClassAValidator()
        {
            RuleFor(x => x.Text1)
                .NotEmpty();

            RuleFor(x => x.Text2)
                .NotEmpty()
                .Equal(x => x.Text1);
        }
}
```

In my View : 

```
<input asp-for="ClassA.Text1" class="form-control" type="text">
<span asp-validation-for="ClassA.Text1" class="text-danger"></span>

<input asp-for="ClassA.Text2" class="form-control" type="text">
<span asp-validation-for="ClassA.Text2" class="text-danger"></span>
```

Going on the html output, I have observed the following difference with Asp.net Core built in validations : 

- FluentValidation : 
`<div ...... data-val-equalto-other="Text1" ....>`

- Asp.net Core : 
`<div ...... data-val-equalto-other="*.Text1" ....>`

In the first case, the validation is not executed by jquery validate because there is no matching with  "Text1".
In the second case, the validation is executed by jquery validate because there is a matching with "*.Text1".

Thanks.

**Comment by JeremySkinner** [comment]
Actually a quick glance at 
https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.DataAnnotations/CompareAttributeAdapter.cs shows the DataAnnotations adapter hard-codes the “*.”, which is missing from the FV equivalent. I’ll try and get a fix published tomorrow.

**Comment by ozerty** [comment]
That's great !!!
Thanks Jeremy

**Comment by JeremySkinner** [comment]
I've released 8.0.101 with a fix for this. Please give it a try and let me know if you run into any issues.
